src/components/Transaction.tsx

```diff
@@ -15,17 +15,38 @@ const source = /* GraphQL */ `
     }
 `;
 
-export default function Transaction() {
-    const [data, setData] = useState();
+/**
+ * Component GraphQL data.
+ */
+type Data = {
+    blockTime: bigint,
+    message: TransactionMessage,
+    signatures: [Signature],
+    slot: Slot,
+    version: String
+};
+
+interface Props {
+    address: Address
+}
+
+export default function Transaction(props: Props) {
+    const [data, setData] = useState<Array<Data>>();
 
     useEffect(() => {
-        const fetchData = async () => {
-            let signature = '67rSZV97NzE4B4ZeFqULqWZcNEV2KwNfDLMzecJmBheZ4sWhudqGAzypoBCKfeLkKtDQBGnkwgdrrFM8ZMaS3pkk' as Signature;
-            const response = (await gql(source, { signature }));
-            console.log(response)
-            setData(response.account);
+        const fetchSignaturesAndLoadTransactions = async () => {
+            const body = { address: props.address };
+            let signatures = await _fetch('/api/getSignaturesForAddress',body) as Array<Signature>;
+
+            const transactions:Array<Data> = [];
+            signatures.forEach(async (signature) => {
+                const response = (await gql(source, { signature:signature })) as { transaction: Data };
+                console.log(response)
+            })
+
+            setData(transactions);
```

Comment:
This is in the wrong place. The `Transactions` _page_ should be making this call once, to gather all the signatures for the wallet address, and then using each signature as a property input on a list of `Transaction` components.